extensions/Elasticsearch/Elasticsearch.FunctionalTests/Additional/KernelMemoryTests.cs

```diff
@@ -15,12 +15,29 @@ public class KernelMemoryTests : MemoryDbFunctionalTest
     public KernelMemoryTests(IConfiguration cfg, ITestOutputHelper output)
         : base(cfg, output)
     {
-        this.KernelMemory = new KernelMemoryBuilder()
-            .With(new KernelMemoryConfig { DefaultIndexName = "default4tests" })
-            .WithSearchClientConfig(new SearchClientConfig { EmptyAnswer = NotFound })
-            .WithOpenAI(this.OpenAiConfig)
-            .WithElasticsearchMemoryDb(this.ElasticsearchConfig)
-            .Build<MemoryServerless>();
+        if (cfg.GetValue<bool>("UseAzureOpenAI"))
+        {
+            Assert.False(string.IsNullOrEmpty(this.AzureOpenAIEmbeddingConfiguration.APIKey));
```

Comment:
this should check for the type of auth. When using managed identities, the API key is empty